### PR TITLE
#171911664: Improve test coverage on app.js

### PR DESCRIPTION
**Pull request by Kagorora**
#### What does this PR do?

- This PR will improve the test coverage by testing app.js

#### Description of Task to be completed?

- will test sockect.io , It's connection and the disconnecting lines

#### How should this be manually tested?

- run `git clone https://github.com/andela/knights-bn-backend.git` to clone the repo.

- run `cd knights-bn-backend` in your terminal to enter into the repository.

- create two Postgres database for development and then test

- run `npm run createTables` to create tables for you

- run `npm install` to install dependencies.

- create a .env file then add environment variable as shown in ` .env.example`,  check our slack channel `code-knights-12` for .env

- run `npm run test` to test.

or 

   consult the badge `https://coveralls.io/github/andela/knights-bn-backend?branch=develop`

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?

- [171911664](https://www.pivotaltracker.com/story/show/171911664)

#### Screenshots (if appropriate)

- N/A

#### Questions:

- N/A

### CODE REVIEW

File: src/controllers/request.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ export default class requestsController {
         where: { requesterId: req.user.id },
         include: [{ model: models.Comment, attributes: ['id', 'comment', 'createdAt'] }],
       });
-      if (allMyRequest) { return res.status(200).json({ message: 'List of requests', allMyRequest }); }
+      if (allMyRequest) { return res.status(200).json({ message: 'all my list of requests', allMyRequest }); }
```
Comment (kagabof) [inline src/controllers/request.js]: Why did you change the message?

File: src/controllers/request.js [hunk 2]
```diff
@@ -114,7 +114,8 @@ export default class requestsController {
         where: { managerId: req.user.id, status: 'pending' },
         include: [{ model: models.Comment }],
       });
-      if (pendingRequests) { return res.status(200).json({ message: 'Pending requests', pendingRequests }); }
+      const pendingRequestLength = pendingRequests.length;
```
Comment (kagabof) [inline src/controllers/request.js:117]: Why did you add this line?

File: package.json [hunk 3]
```diff
@@ -62,6 +63,7 @@
     "sequelize": "^5.21.3",
     "sequelize-cli": "^5.5.1",
     "socket.io": "^2.3.0",
+    "socket.io-client": "^2.3.0",
```
Comment (kagabof) [inline package.json:66]: I don't understand why you are adding this "socket.io-client", Please explain.

**Review by kagabof — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (Kagorora) [reply package.json:66]: socket.io-client is the for the client-side implementation of socket.io, I have introduced it to allow me to simulate the socket that the other user(client uses) in order to be able to test as I need a socket to be active to be able to connect .

File: src/controllers/request.js (hunk 2 above)
Comment (Kagorora) [reply src/controllers/request.js:117]: This line was there before it was added to avoid this code maintainability error ('Similar blocks of code')

File: src/controllers/request.js (hunk 1 above)
Comment (Kagorora) [reply src/controllers/request.js]: I found out that there was no need to add that line so I have removed it

**Review by Moise1 — CHANGES_REQUESTED** [review]
@Kagorora, here is my comment on PR: 

1. Please make a proper PR description. 
2. Please name your commit message properly. i.e.: Avoid uppercase letters in the PR's head.
3. Fix your Travis CI build that's failing.

**Review by Moise1 — APPROVED** [review]

**Review by Niyonsengaeric — APPROVED** [review]

File: src/app.js [hunk 4]
```diff
@@ -22,24 +22,24 @@ dotenv.config();
...
-      const clientKey = Number.parseInt(userData.id, 10);
-      connectedClients[clientKey] = connectedClients[clientKey] || [];
-      connectedClients[clientKey].push(socket.id);
-    }
+    const clientKey = Number.parseInt(userData.id, 10);
+    connectedClients[clientKey] = connectedClients[clientKey] || [];
+    connectedClients[clientKey].push(socket.id);
+    next();
+  } catch (error) {
+    console.log('Invalid token provided');
```
Comment (T2Wil) [inline src/app.js:34]: hey @Kagorora , isn't this supposed to be returned as a response?

Comment (Kagorora) [reply src/app.js:34]: @T2Wil  when returning error in the catch, I can't see that message that's why I have consoled this

Comment (kagabof) [reply src/app.js:34]: What do you mean when you say you can't see that message?

**Review by kagabof — APPROVED** [review]
